### PR TITLE
Troubleshooting guidance for newly added custom checks

### PR DESCRIPTION
**Pull request by ramonsmits**

### CODE REVIEW

File: servicecontrol/troubleshooting.md [hunk 1]
```diff
@@ -116,6 +116,30 @@ Although the installer is code signed correctly with a certificate owned by "NSe
...
+
+The database technology used for ServiceControl is based on asynchonous index updates. Indexes are not updated immediately but very soon after data updates. When a system is healhty indexes are updated in milliseconds or several seconds under load. When indexes get very stale this means that indexes lag behind for a certain duration. 
+
+Systems are affected by severe index lag when the following custom check message is presented:
+
+> At least one index significantly stale. Please run maintenance mode if this custom check persists to ensure index(es) can recover. See log file in `{LogPath}` for more details. Visit https://docs.particular.net/search?q=servicecontrol+troubleshooting for more information.
+
+By launching an instance in maintenance mode the message ingestion halts but the database . This ensures that any tasks related to index rebuilding or index scanning can run without interruption. This is useful when the storage isn't fast enough to do both message ingestion and index operations.
+
+Consider upgrading the storage if this errors persists.
```
Comment (ramonsmits) [inline servicecontrol/troubleshooting.md:129]:
```suggestion
Consider upgrading the storage if these errors persists.
```

File: servicecontrol/troubleshooting.md [hunk 2]
```diff
@@ -116,6 +116,30 @@ Although the installer is code signed correctly with a certificate owned by "NSe
 
 When building ServiceControl, all build artifacts are virus scanned to ensure no viruses or malware are shipped with the installer packages.
 
+## Stale indexes 
+
+The database technology used for ServiceControl is based on asynchonous index updates. Indexes are not updated immediately but very soon after data updates. When a system is healhty indexes are updated in milliseconds or several seconds under load. When indexes get very stale this means that indexes lag behind for a certain duration.
```
Comment (ramonsmits) [inline servicecontrol/troubleshooting.md:121]:
```suggestion
The database technology used for ServiceControl is based on asynchronous index updates. Indexes are not updated immediately, but very soon after data updates. A healthy system has indexes updated in milliseconds or several seconds under load. When indexes get very stale this means that indexes lag behind for a long duration and can affect data presented and started tasks.
```

File: servicecontrol/troubleshooting.md [hunk 3]
```diff
@@ -116,6 +116,30 @@ Although the installer is code signed correctly with a certificate owned by "NSe
...
 
+## Stale indexes 
+
+The database technology used for ServiceControl is based on asynchonous index updates. Indexes are not updated immediately but very soon after data updates. When a system is healhty indexes are updated in milliseconds or several seconds under load. When indexes get very stale this means that indexes lag behind for a certain duration. 
+
+Systems are affected by severe index lag when the following custom check message is presented:
+
+> At least one index significantly stale. Please run maintenance mode if this custom check persists to ensure index(es) can recover. See log file in `{LogPath}` for more details. Visit https://docs.particular.net/search?q=servicecontrol+troubleshooting for more information.
+
+By launching an instance in maintenance mode the message ingestion halts but the database . This ensures that any tasks related to index rebuilding or index scanning can run without interruption. This is useful when the storage isn't fast enough to do both message ingestion and index operations.
```
Comment (ramonsmits) [inline servicecontrol/troubleshooting.md:127]:
```suggestion
By launching an instance in maintenance mode the message ingestion stops but database engine still runs. This ensures that any tasks related to index rebuilding or index scanning can run without interruption. This is useful when the storage isn't fast enough to do both message ingestion and index operations.
```

File: servicecontrol/troubleshooting.md [hunk 4]
```diff
@@ -116,6 +116,30 @@ Although the installer is code signed correctly with a certificate owned by "NSe
...
+
+Consider upgrading the storage if this errors persists.
+
+Contact [support](https://particular.net/support) for assistance
```
Comment (ramonsmits) [inline servicecontrol/troubleshooting.md:131]:
```suggestion
Contact [support](https://particular.net/support) for assistance.
```

File: servicecontrol/troubleshooting.md [hunk 5]
```diff
@@ -116,6 +116,30 @@ Although the installer is code signed correctly with a certificate owned by "NSe
...
+
+## Index errors
+
+Index issues usually are automatically corrected at start but sometimes index issues require manual intervention. When unrecoverable index issues occur the following custom check message is visible:
+
+> Detected RavenDB index errors, please start maintenance mode and resolve the following issues:
+
+Most often issue is that [indexes get corrupted](#indexes-get-corrupted). Resolve these errors by inspecting these errors via [ServiceControl (audit or error) maintenance mode](maintenance-mode.md).
+
+Contact [support](https://particular.net/support) for assistance
```
Comment (ramonsmits) [inline servicecontrol/troubleshooting.md:141]:
```suggestion
Contact [support](https://particular.net/support) for assistance.
```

**Comment by ramonsmits** [comment]
@DavidBoike is there anyway to ignore the content verification check? The text is a 1-1 copy from the message in ServicePulse which I do not want to remove.

**Comment by DavidBoike** [comment]
Not at the moment, but it's a good idea to add a feature to be able to pragma something like that out somehow.

File: servicecontrol/troubleshooting.md [hunk 6]
```diff
@@ -116,6 +116,30 @@ Although the installer is code signed correctly with a certificate owned by "NSe
 
 When building ServiceControl, all build artifacts are virus scanned to ensure no viruses or malware are shipped with the installer packages.
 
+## Stale indexes 
+
+The database technology used for ServiceControl is based on asynchronous index updates. Indexes are not updated immediately, but very soon after data updates. A healthy system has indexes updated in milliseconds or several seconds under load. When indexes get very stale this means that indexes lag behind for a long duration and can affect data presented and started tasks.
+
+Systems are affected by severe index lag when the following custom check message is presented:
+
+> At least one index significantly stale. Please run maintenance mode if this custom check persists to ensure index(es) can recover. See log file in `{LogPath}` for more details. Visit https://docs.particular.net/search?q=servicecontrol+troubleshooting for more information.
```
Comment (ramonsmits) [inline servicecontrol/troubleshooting.md:125]:
```suggestion
> At least one index significantly stale. Please run maintenance mode if this custom check persists to ensure index(es) can recover. See log file in `{LogPath}` for more details.
```

**Comment by ramonsmits** [comment]
> Not at the moment, but it's a good idea to add a feature to be able to pragma something like that out somehow.

Ok, dropping the sentence containing the url. Created a issue for the pragma directive: https://github.com/Particular/DocsEngine/issues/621

**Pull request merged**